LibOS and Pal don't build on Ubuntu 17.10

I tried to build Graphene on Ubuntu 17.10, but it failed with the following errors:

### LibOS
```
graphene/LibOS$ make
Building glibc, may take a while to finish. Warning messages may show up. If this process terminates with failures, see "glibc-build/build.log" for more information.
cd glibc-build && ((make >> build.log) 2>&1 | tee -a build.log)
/usr/bin/ld: graphene/LibOS/glibc-build/csu/crt1.o: relocation R_X86_64_32S against symbol `__libc_csu_fini' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/crtbegin.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when making a shared object
/usr/bin/ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
make[3]: *** [graphene/LibOS/glibc-build/iconv/iconvconfig] Error 1
make[2]: *** [iconv/others] Error 2
make[1]: *** [all] Error 2
make -C shim all
make[1]: Entering directory 'graphene/LibOS/shim'
make all -C src
make[2]: Entering directory 'graphene/LibOS/shim/src'
[ shim_parser.o ]
In file included from ../include/shim_internal.h:66:0,
                 from shim_parser.c:27:
shim_parser.c: In function ‘parse_syscall_args’:
shim_parser.c:451:40: error: first argument to ‘va_arg’ not of type ‘va_list’
     const char * arg_type = va_arg(ap, const char *);
                                        ^
shim_parser.c: In function ‘skip_syscall_args’:
shim_parser.c:463:41: error: first argument to ‘va_arg’ not of type ‘va_list’
     const char * arg_type = va_arg (ap, const char *);
                                         ^
shim_parser.c:466:20: error: first argument to ‘va_arg’ not of type ‘va_list’
         va_arg(ap, const char *);
                    ^
shim_parser.c:468:20: error: first argument to ‘va_arg’ not of type ‘va_list’
         va_arg(ap, void *);
                    ^
shim_parser.c:470:20: error: first argument to ‘va_arg’ not of type ‘va_list’
         va_arg(ap, int);
                    ^
Makefile:109: recipe for target 'shim_parser.o' failed
make[2]: *** [shim_parser.o] Error 1
make[2]: Leaving directory 'graphene/LibOS/shim/src'
Makefile:13: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory 'graphene/LibOS/shim'
Makefile:13: recipe for target 'all' failed
make: *** [all] Error 2
```
### Pal
```
graphene/Pal/src$ make
Building target: Linux
make -C ../lib target=graphene/Pal/src/.lib/
make[1]: Entering directory 'graphene/Pal/lib'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory 'graphene/Pal/lib'
make[1]: Entering directory 'graphene/Pal/src/host/Linux'
[ host/Linux/db_pipes.o ]
In file included from /usr/include/x86_64-linux-gnu/bits/types/sigset_t.h:4:0,
                 from /usr/include/x86_64-linux-gnu/sys/select.h:33,
                 from /usr/include/x86_64-linux-gnu/sys/types.h:197,
                 from /usr/include/x86_64-linux-gnu/bits/socket.h:29,
                 from /usr/include/x86_64-linux-gnu/sys/socket.h:33,
                 from db_pipes.c:42:
/usr/include/x86_64-linux-gnu/bits/types/__sigset_t.h:8:3: error: conflicting types for ‘__sigset_t’
 } __sigset_t;
   ^~~~~~~~~~
In file included from pal_linux.h:32:0,
                 from db_pipes.c:31:
include/sigset.h:36:5: note: previous declaration of ‘__sigset_t’ was here
   } __sigset_t;
     ^~~~~~~~~~
Makefile:43: recipe for target 'db_pipes.o' failed
make[1]: *** [db_pipes.o] Error 1
make[1]: Leaving directory 'graphene/Pal/src/host/Linux'
Makefile:94: recipe for target 'host/Linux/libpal-Linux.a' failed
make: *** [host/Linux/libpal-Linux.a] Error 2
```

Thanks for the fix, but it still doesn't compile. The error is the same:
```
[ host/Linux/db_pipes.o ]
In file included from /usr/include/x86_64-linux-gnu/bits/types/sigset_t.h:4:0,
                 from /usr/include/x86_64-linux-gnu/sys/select.h:33,
                 from /usr/include/x86_64-linux-gnu/sys/types.h:197,
                 from /usr/include/x86_64-linux-gnu/bits/socket.h:29,
                 from /usr/include/x86_64-linux-gnu/sys/socket.h:33,
                 from db_pipes.c:42:
/usr/include/x86_64-linux-gnu/bits/types/__sigset_t.h:8:3: error: conflicting types for ‘__sigset_t’
 } __sigset_t;
   ^~~~~~~~~~
In file included from pal_linux.h:32:0,
                 from db_pipes.c:31:
include/sigset.h:36:5: note: previous declaration of ‘__sigset_t’ was here
   } __sigset_t;
     ^~~~~~~~~~
Makefile:43: recipe for target 'db_pipes.o' failed
```

--- a/LibOS/shim/src/shim_parser.c
+++ b/LibOS/shim/src/shim_parser.c
@@ -448,7 +448,7 @@ static inline void parse_integer_arg (va_list * ap)
 
 static inline void parse_syscall_args (va_list * ap)
 {
-    const char * arg_type = va_arg(ap, const char *);
+    const char * arg_type = va_arg(*ap, const char *);
 
     if (strcmp_static(arg_type, "const char *"))
         parse_string_arg(ap);
@@ -460,14 +460,14 @@ static inline void parse_syscall_args (va_list * ap)
 
 static inline void skip_syscall_args (va_list * ap)
 {
-    const char * arg_type = va_arg (ap, const char *);
+    const char * arg_type = va_arg(*ap, const char *);
 
     if (strcmp_static(arg_type, "const char *"))
-        va_arg(ap, const char *);
+        va_arg(*ap, const char *);
     else if (is_pointer(arg_type))
-        va_arg(ap, void *);
+        va_arg(*ap, void *);
     else
-        va_arg(ap, int);
+        va_arg(*ap, int);
 }
 
 void sysparser_printf (const char * fmt, ...)

